pages/emotion/[id]/edit.js

```diff
@@ -1,20 +1,23 @@
 import EmotionForm from "@/components/EmotionForm";
 import Heading from "@/components/Heading";
 import { useRouter } from "next/router";
+import useSWR from "swr";
 
-export default function EditPage({
-  emotions,
-  onUpdateEmotion,
-  customEmotionTypes,
-}) {
+export default function EditPage({ onUpdateEmotion }) {
   const router = useRouter();
   const { id } = router.query;
 
-  const existingEmotion = emotions.find((emotion) => emotion.id === id);
-  if (!existingEmotion) {
-    return <p>Loading...</p>;
+  const { data: emotions, error, isLoading } = useSWR("/api/emotionEntries");
```

Comment:
here you should send a GET request to your other "emotionEntries" endpoint - the one that that returns only one element: `/api/emotionEntries/${id}`. You don't need to `find` the correct emotion below anymore, since that was done by the database.